letta/llm_api/openai.py

```diff
@@ -533,7 +536,10 @@ def openai_chat_completions_request(
 
     if "tools" in data:
         for tool in data["tools"]:
-            tool["function"] = convert_to_structured_output(tool["function"])
+            try:
+                tool["function"] = convert_to_structured_output(tool["function"])
+            except ValueError as e:
```

Comment:
note @mattzh72 in cases where the user made a tool that has a `dict` for example (which can't convert to structurd outputs format), we catch the error here and use non-SO schema instead